src/OpenTracing.Noop/NoopScopeManager.cs

```diff
@@ -0,0 +1,46 @@
+namespace OpenTracing.Noop
+{
+    /// <summary>
+    /// A noop (i.e., cheap-as-possible) implementation of an <see cref="IScopeManager"/>.
+    /// </summary>
+    public sealed class NoopScopeManager : IScopeManager
+    {
+        internal static readonly NoopScopeManager Instance = new NoopScopeManager();
+
+        public IScope Active => null;
+
+        private NoopScopeManager()
+        {
+        }
+
+        public IScope Activate(ISpan span, bool finishSpanOnClose)
+        {
+            return NoopScope.Instance;
+        }
+
+        public override string ToString()
+        {
+            return GetType().Name;
+        }
+
+        public sealed class NoopScope : IScope
+        {
+            internal static readonly NoopScope Instance = new NoopScope();
+
+            public ISpan Span => NoopSpan.Instance;
+
+            private NoopScope()
+            {
+            }
+
+            public void Dispose()
+            {
+            }
+
+            public override string ToString()
+            {
+                return GetType().Name;
```

Comment:
If we want to be "cheap-as-possible"
> A noop (i.e., cheap-as-possible) implementation of an <see cref="IScopeManager"/>.

we should use constants here, which are safe since these classes are sealed.

``` ini

BenchmarkDotNet=v0.10.12, OS=Windows 10 Redstone 3 [1709, Fall Creators Update] (10.0.16299.192)
Intel Core i7-6820HQ CPU 2.70GHz (Skylake), 1 CPU, 8 logical cores and 4 physical cores
Frequency=2648437 Hz, Resolution=377.5812 ns, Timer=TSC
  [Host]     : .NET Framework 4.7 (CLR 4.0.30319.42000), 32bit LegacyJIT-v4.7.2600.0
  DefaultJob : .NET Framework 4.7 (CLR 4.0.30319.42000), 32bit LegacyJIT-v4.7.2600.0


```
|  Method |       Mean |     Error |    StdDev | Allocated |
|-------- |-----------:|----------:|----------:|----------:|
| GetType | 14.4007 ns | 0.3537 ns | 0.4210 ns |       0 B |
|  TypeOf | 13.0086 ns | 0.3230 ns | 0.2864 ns |       0 B |
|   Const |  0.6814 ns | 0.0805 ns | 0.0714 ns |       0 B |

Reply:
Agreed. I'll use `nameof(...)` instead (which is a compile-time constant)!